client/src/PdfGenerator.test.js

```diff
@@ -0,0 +1,22 @@
+import React from "react";
+import { describe, expect, it } from "@jest/globals";
+import { shallow, mount } from "enzyme";
+import toJson from "enzyme-to-json";
+import PdfGenerator from "./PdfGenerator";
+
+describe("PdfGenerator", () => {
+    it("renders", () => {
+        const wrapper = shallow(<PdfGenerator />);
+        expect(toJson(wrapper)).toMatchSnapshot();
+    });
+
+    it("changes the pid input", () => {
+        const wrapper = mount(<PdfGenerator />);
+        wrapper.find("#pdfGeneratePid").simulate("change", {
+            target: {
+                value: "testPid",
+            },
+        });
+        expect(wrapper.find("#pdfGeneratePid").instance().value).toEqual("testPid");
```

Comment:
Is this test actually testing any of the behavior of the React component itself? It looks to me like you're setting the value of an input, and then confirming that the value of the input changed... but that doesn't seem to confirm whether any of the related React logic actually did anything. As always, apologies if I'm missing the point here. :-)

Reply:
I send an onChange event in the previous line. After onChange, it should call setPid and update pid state. After update of pid state, on the next render because of state change, pid value is passed to the input value attribute into input.

Reply:
Okay, great, that makes sense -- thanks for the clarification!